chronos/transports/stream.nim

```diff
@@ -2112,6 +2126,34 @@ template fastWrite(transp: auto, pbytes: var ptr byte, rbytes: var int,
             retFuture.fail(error)
             return retFuture
 
+proc addToWriteQueue(transp: StreamTransport, vector: StreamVector) =
+  transp.queue.addLast(vector)
+
+  # to catch cancellation
+  let childFut = newFuture[void]("stream.transport.write.child")
```

Comment:
This new future is a bit unfortunate, but this the current Futures system, this is the only way to catch a cancel request, and be able to swallow it (which will happen here if we can't cancel)
I think it's useful to let higher level know if their message did get sent or not, but if we don't care about that, we can just remove this